WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift

```diff
@@ -862,12 +868,33 @@ private extension NotificationsViewController {
         NotificationSyncMediator()?.markAsRead(note)
     }
 
+    /// Marks all messages as read under the selected filter.
+    ///
+    @objc func markAllAsRead() {
+        guard let notes = tableViewHandler.resultsController.fetchedObjects as? [Notification] else {
+            return
+        }
+
+        let unreadNotifications = notes.filter {
+            !$0.read
+        }
+
+        NotificationSyncMediator()?.markAsRead(unreadNotifications)
+        let notice = Notice(
+            title: Localization.markAllAsReadNoticeTitle,
+            message: Localization.markAllAsReadNoticeMessage
+        )
+        ActionDispatcherFacade().dispatch(NoticeAction.post(notice))
+        updateMarkAllAsReadTintColor()
```

Comment:
There's a few things we can improve here.
1. Move these into `markAsRead`'s `completion` block so they are only triggered if there is not an error.
2. If there is an error, dispatch a failed notice instead. Something like `Failed marking Notifications as read`.
3. I think in this case we don't need both a notice title and message. I think simply `Notifications marked as read` or `Failed marking Notifications as read` is sufficient. (Notice there are no periods at the end of these messages.)